Compiler/src/Compiler.cpp

```diff
@@ -1477,6 +1480,70 @@ struct Compiler
         }
     }
 
+    void compileExprInterpString(AstExprInterpString* expr, uint8_t target, bool targetTemp)
+    {
+        std::string formatString;
+
+        size_t stringsLeft = expr->strings.size;
+
+        for (AstArray<char> const& string : expr->strings)
+        {
+            std::string stringEscaped(string.data);
```

Comment:
Extra allocation here plus the loop below is quadratic (a string made of 10k '%' takes excessive amount of time to process). It would be better to go through the source string and append character by character.

Reply:
Plus maybe a fast-path where if a string doesn't contain a character '%' (can be checked via `memchr`), you simply append, otherwise use a character-by-character loop that appends one-two characters at a time.